colibre/scripts/bh_luminosities.py

```diff
@@ -0,0 +1,172 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.stats as stats
+
+from swiftsimio import load
+
+import unyt
+from unyt import mh, cm, Gyr, speed_of_light
+from matplotlib.colors import LogNorm
+from matplotlib.animation import FuncAnimation
+from matplotlib import rcParams
+
+# Set the limits of the figure.
+mass_bounds = [1e5, 3e10]
+value_bounds = [1e38, 1e48]
+bins = 25
+def_value = -1.0
+plot_scatter = True
+
+
+def get_data(filename):
+
+    data = load(filename)
+
+    masses = data.black_holes.subgrid_masses.to("Msun")
+
+    try:
+        rad_effs = data.black_holes.radiative_efficiencies
+        accr_rates = data.black_holes.accretion_rates.astype(np.float64).to(
+            unyt.kg / unyt.s
+        )
+        values = rad_effs * (speed_of_light ** 2) * accr_rates
+        values = values.to(unyt.erg / unyt.s)
+
+        # Make sure to take only black holes with a non-zero luminosity (efficiency)
+        masses = masses[rad_effs > 1e-6]
+        values = values[rad_effs > 1e-6]
+    except:
+        values = unyt.unyt_array(
+            np.zeros(masses.shape), dtype=np.float64, units=unyt.erg / unyt.s
+        )
+
+    return masses, values
+
+
+def calculate_medians(filename, mass_bounds, value_bounds, bins):
```

Comment:
The argument `value_bounds` is not used inside the function

Reply:
Removed now in all the scripts.